lib/backpex/html/form.ex

```diff
@@ -4,152 +4,249 @@ defmodule Backpex.HTML.Form do
   """
   use BackpexWeb, :html
 
-  alias Phoenix.HTML.Form, as: PhoenixForm
+  alias Phoenix.HTML.Form
 
   @doc """
   Renders an input.
   """
   @doc type: :component
 
-  attr(:form, :atom, required: true, doc: "the form")
-  attr(:field_name, :atom, required: true, doc: "the field name")
-  attr(:field_options, :map, required: true, doc: "the field options")
-  attr(:options, :list, doc: "the options to be used for a select input")
-  attr(:value, :any, doc: "the value of the form input")
-
-  attr(:type, :string,
-    required: true,
-    values: ~w(text textarea toggle number date datetime-local select)
-  )
-
-  attr(:rest, :global, include: ~w(readonly disabled min step))
+  attr :id, :any, default: nil
+  attr :name, :any
+  attr :label, :string, default: nil
+  attr :value, :any
+
+  attr :type, :string,
+    default: "text",
+    values: ~w(checkbox color date datetime-local email file hidden month number password
+               range radio search select tel text textarea time url week toggle)
+
+  attr :field, Phoenix.HTML.FormField, doc: "a form field struct retrieved from the form, for example: @form[:email]"
+
+  attr :errors, :list, default: []
+  attr :checked, :boolean, doc: "the checked flag for checkbox inputs"
+  attr :prompt, :string, default: nil, doc: "the prompt for select inputs"
+  attr :options, :list, doc: "the options to pass to Phoenix.HTML.Form.options_for_select/2"
+  attr :multiple, :boolean, default: false, doc: "the multiple flag for select inputs"
+
+  attr :rest, :global, include: ~w(accept autocomplete capture cols disabled form list max maxlength min minlength
+                multiple pattern placeholder readonly required rows size step)
+
+  attr :class, :string, default: nil, doc: "additional class"
+  attr :input_class, :string, default: nil, doc: "additional class for the input element"
+
+  attr :input_wrapper_class, :string,
+    default: nil,
+    doc: "additional class for the input wrapper element, currently only used in select type"
+
+  attr :field_options, :map, default: %{}, doc: "field options map"
+  attr :hide_errors, :boolean, default: false, doc: "if errors should be hidden"
+
+  slot :inner_block
+
+  def field_input(%{field: %Phoenix.HTML.FormField{} = field} = assigns) do
+    translated_errors =
+      field.errors
+      |> Enum.map(fn error ->
+        {text, _error} = translate_form_error(error, assigns.field_options)
+        text
+      end)
```

Comment:
We also have this code in `multi_select`. Maybe add a `translated_form_errors` function?